$DragonFly: src/gnu/usr.bin/cc34/cc_prep/patches/gcse.c.patch,v 1.3 2004/12/21 13:10:48 joerg Exp $

===================================================================
RCS file: /home/joerg/wd/repository/dragonflybsd/src/contrib/gcc-3.4/gcc/gcse.c,v
retrieving revision 1.2
diff -u -r1.2 gcse.c
--- gcse.c	20 Dec 2004 19:23:24 -0000	1.2
+++ gcse.c	20 Dec 2004 20:48:10 -0000
@@ -4178,7 +4178,7 @@
       /* Find an assignment that sets reg_used and is available
 	 at the start of the block.  */
       set = find_avail_set (regno, insn);
-      if (! set)
+      if (! set || SET_VOLATILE_P (set->expr))
 	continue;
 
       pat = set->expr;
